### PR TITLE
[ISSUE #6226] Shutdown flowMonitor when connection disconnect

### LINKED ISSUE #6226: a large number of FlowMonitor thread in rocketmq
use the default config to start up a rocketmq cluster,when it run for a month or so, there will be a jvm crash because of oom. we try to modify the linux memory arguments , and that is not worked. and we get the jvm dump and thread stack. now we found that a lot of FlowMonitor  thread  wating, and it keep increase.we suspect the amount of flowmonitor thread is the reason of jvm crash.  

please read the thread analysis and jvm crash, and tell us the true reason why it crash, and the way to fix it, thanks!

 environment: 
rocketmq 5.0.0
linux redhat7.9, 48Core/755GB MEM  

/usr/local/jdk/bin/java -server -Xms8g -Xmx8g -XX:+UseG1GC -XX:G1HeapRegionSize=16m -XX:G1ReservePercent=25 -XX:InitiatingHeapOccupancyPercent=30 -XX:SoftRefLRUPolicyMSPerMB=0 -verbose:gc -Xloggc:/dev/shm/rmq_srv_gc_%p_%t.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintAdaptiveSizePolicy -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=30m -XX:-OmitStackTraceInFastThrow -XX:+AlwaysPreTouch -XX:MaxDirectMemorySize=15g -XX:-UseLargePages -XX:-UseBiasedLocking -Drocketmq.client.logUseSlf4j=true -cp .:/data/rocketmq-all-5.0.0-bin-release//bin/../conf:/data/rocketmq-all-5.0.0-bin-release//bin/../lib/*: -Djdk.tls.rejectClientInitiatedRenegotiation=true org.apacherocketmq.broker.BrokerStartup -c /data/rocketmq/conf/broker-c.properties

the jvm crash info:
image

thread dump:
https://fastthread.io/my-thread-report.jsp?p=c2hhcmVkLzIwMjMvMDMvMi90aHJlYWQudHh0LS03LTE0LTU2&

### PR DESCRIPTION
**Pull request by RongtongJin**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

fix #6226

## Brief changelog

Shutdown flowMonitor when connection disconnect

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by RongtongJin** [comment]
@guyinyou PTAL～

**Review by guyinyou — APPROVED** [review]
lgtm

**Review by guyinyou — DISMISSED** [review]

File: controller/src/main/java/org/apache/rocketmq/controller/impl/manager/SyncStateInfo.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ public void updateSyncStateSetInfo(Set<String> newSyncStateSet) {
     }
 
     public boolean isMasterExist() {
-        return !this.masterAddress.isEmpty();
+        return masterAddress != null && !this.masterAddress.isEmpty();
```
Comment (ShannonDing) [inline controller/src/main/java/org/apache/rocketmq/controller/impl/manager/SyncStateInfo.java:56]: StringUtils.isNotBlank()?

**Review by ShannonDing — APPROVED** [review]

**Review by Focus-rth — APPROVED** [review]
lgtm

**Pull request merged**